### PR TITLE
feat: enable better cssOptions override for modules

### LINKED ISSUE #29955: gatsby-plugin-sass prevents configuration of CSS Loader
## Description

When setting the following options to `gatsby-config.js`: 

````js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-sass`,
      options: {
        cssLoaderOptions: {
          modules: {
            exportLocalsConvention: 'camelCaseOnly'
          }
        }
      }
    }
  ]
}
````

We expect the exports of `css-loader` to be `camelCaseOnly` and not `dashesOnly` (default).

### Steps to reproduce

Very straightforward, just place this in a Gatsby SASS project.

### Expected result

The webpack configuration should be modified accordingly and the exported values should be in camel case.

### Actual result

Nothing changes. After investigation, it seems that the culprit is in `gatsby-plugin-sass/gatsby-node.js` where there is the following:

````js
const sassRuleModules = {
    test: sassRuleModulesTest || /\.module\.s(a|c)ss$/,
    use: [!isSSR && loaders.miniCssExtract({
      modules: true
    }), loaders.css({
      importLoaders: 2,
      ...cssLoaderOptions,
      modules: true
    }), loaders.postcss({
      plugins: postCssPlugins
    })].filter(Boolean)
  };
````

This causes the modules option to be overwritten even if it was provided. There should be a check to see if there is already a modules option to prevent overwriting.

### Environment

````
  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  Binaries:
    Node: 15.3.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.9 - ~\AppData\Roaming\npm\npm.CMD
  Languages:
    Python: 3.6.4
  Browsers:
    Chrome: 88.0.4324.190
    Edge: Spartan (44.19041.423.0), Chromium (88.0.705.81)
  npmPackages:
    gatsby: ^3.0.0 => 3.0.0
    gatsby-plugin-mdx: ^2.0.0 => 2.0.0
    gatsby-plugin-sass: ^4.0.0 => 4.0.0
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-telemetry: ^2.0.0 => 2.0.0
  npmGlobalPackages:
    undefined: 0.1.0
````

### LINKED ISSUE #30040: [Gatsby v3] CSS Modules - Default exports from dependency lib cause broken build
## Description

1. **Spike**: Following the migration build, I'm certain this is caused from [css-modules-are-imported-as-es-modules](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#css-modules-are-imported-as-es-modules).

2. **Problem**: When importing from an external component lib that uses default exports for `scss`, we see the warning
```javascript
warn Attempted import error: './styles/FormError.module.scss' does not contain a default export (imported as 'styles').
```

3. **Question**: **Do we have to wait until our libs that are using default exports like** [`import styles from './styles.module.scss';`](https://github.com/Codecademy/client-modules/blob/546c453575b41c4891e0781a217a05f381f25aac/packages/gamut/src/SkipToContent/index.tsx#L5) **are updated to use** `named export`**, or is there a work around until then?**

### Steps to reproduce
1. Clone https://github.com/rogervera/gatsby-v3-default-export-bug-repro
2. `gatsby develop`
3. See errors on root page.

### Expected result

Auto resolve any runtime errors on `import errors` from dependency packages? 
Not sure what is possible or ideal...

### Actual result

Screen Shot 2021-03-04 at 5 17 13 PM

### Bug Repro Env
```
  System:
    OS: macOS 11.2.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.16.0 - ~/.nvm/versions/node/v12.16.0/bin/node
    Yarn: 1.21.1 - ~/.yarn/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 86.0
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.0.1 => 3.0.2
    gatsby-plugin-sass: ^4.0.1 => 4.0.1
  npmGlobalPackages:
    gatsby: 3.0.2
```

## Actual Env
```
  System:
    OS: macOS 11.2.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.16.0 - ~/.nvm/versions/node/v12.16.0/bin/node
    Yarn: 1.21.1 - ~/.yarn/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 86.0
    Safari: 14.0.3
  npmPackages:
    gatsby: ^3.0.0 => 3.0.1
    gatsby-image: ^2.2.34 => 2.11.0
    gatsby-plugin-alias-imports: ^1.0.5 => 1.0.5
    gatsby-plugin-asset-path: ^3.0.3 => 3.0.3
    gatsby-plugin-canonical-urls: ^3.0.0 => 3.0.0
    gatsby-plugin-emotion: ^6.0.0 => 6.0.0
    gatsby-plugin-newrelic: ^1.0.3 => 1.0.5
    gatsby-plugin-offline: ^4.0.0 => 4.0.0
    gatsby-plugin-postcss: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-resolve-src: ^2.1.0 => 2.1.0
    gatsby-plugin-sass: ^4.0.0 => 4.0.0
    gatsby-plugin-sharp: ^3.0.0 => 3.0.0
    gatsby-plugin-sitemap: ^3.0.0 => 3.0.0
    gatsby-plugin-tsconfig-paths: ^1.0.2 => 1.0.2
    gatsby-plugin-tslint: 0.0.2 => 0.0.2
    gatsby-plugin-typescript: ^3.0.0 => 3.0.0
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4
    gatsby-source-contentful: ^5.0.0 => 5.0.0
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby: 3.0.2
```

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Enable overriding of esModule props to support old commonjs Syntax.
- gatsby-plugin-sass
- gatsby-plugin-less
- gatsby-plugin-postcss
- gatsby-plugin-stylus

### Documentation

PR https://github.com/gatsbyjs/gatsby/pull/30267/ needs to be merged when released

## Related Issues

Fixes #29955
Fixes #30040

[ch27340]

### CODE REVIEW

File: docs/docs/reference/release-notes/migrating-from-v2-to-v3.md [hunk 1]
```diff
@@ -369,6 +369,30 @@ export default Box
 
 You can also still import all styles using the `import * as styles` sytax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
 
+Migrating all your css could be painful. You can still use commonjs with some extra work. This works for sass, less, stylus & regular css.
+This example covers sass but it's the same ofr the other plugins.
```
[the remark this answers is not in the document]
Comment (wardpeet) [reply docs/docs/reference/release-notes/migrating-from-v2-to-v3.md:373]: I'm not sure what you're asking here? We should only enable this through plugins?

[the remark this answers is not in the document]
Comment (LekoArts) [reply docs/docs/reference/release-notes/migrating-from-v2-to-v3.md:373]: We say here:

> You can workaround this issue for Sass, Less, Stylus & **regular CSS modules**

So if they use https://www.gatsbyjs.com/docs/recipes/styling-css#using-css-modules as our default CSS modules functionality we ship by default and they want to override this behavior. How would they do that compared to e.g. the Sass plugin

[the remark this answers is not in the document]
Comment (wardpeet) [reply docs/docs/reference/release-notes/migrating-from-v2-to-v3.md:373]: Use postcss plugin, i'll clarify

File: docs/docs/reference/release-notes/migrating-from-v2-to-v3.md [hunk 2]
```diff
@@ -367,7 +367,30 @@ const Box = ({ children }) => (
 export default Box
 ```
 
-You can also still import all styles using the `import * as styles` sytax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
+You can also import all styles using the `import * as styles` sytax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
```
Comment (LekoArts) [inline docs/docs/reference/release-notes/migrating-from-v2-to-v3.md:370]:
```suggestion
You can also import all styles using the `import * as styles` syntax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
```

File: docs/docs/reference/release-notes/migrating-from-v2-to-v3.md [hunk 3]
```diff
@@ -367,7 +367,7 @@ const Box = ({ children }) => (
 export default Box
 ```
 
-You can also still import all styles using the `import * as styles` syntax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
+You can also still import all styles using the `import * as styles` sytax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
```
Comment (justinwaite) [inline docs/docs/reference/release-notes/migrating-from-v2-to-v3.md:370]:
```suggestion
You can also still import all styles using the `import * as styles` syntax e.g. `import * as styles from './mystyles.module.css'`. However, this won't allow webpack to treeshake your styles so we discourage you from using this syntax.
```

File: packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js [hunk 4]
```diff
@@ -20,6 +20,14 @@ describe(`gatsby-plugin-postcss`, () => {
       "No options": {},
       "PostCss options": { postCssPlugins: [`autoprefixer`], sourceMap: false },
       "Css options": { cssLoaderOptions: { camelCase: true } },
+      "Cssoptions commonjs": {
```
Comment (justinwaite) [inline packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js:23]:
```suggestion
      "Css options commonjs": {
```

File: packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js [hunk 5]
```diff
@@ -76,11 +84,34 @@ describe(`gatsby-plugin-postcss`, () => {
           ],
         },
       }),
+      "Cssoptions commonjs": jest.fn().mockReturnValue({
```
Comment (justinwaite) [inline packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js:87]:
```suggestion
      "Css options commonjs": jest.fn().mockReturnValue({
```

File: packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js [hunk 6]
```diff
@@ -76,11 +84,34 @@ describe(`gatsby-plugin-postcss`, () => {
...
+            },
+          ],
+        },
+      }),
     },
     actions: {
       "No options": actions.setWebpackConfig,
       "PostCss options": actions.replaceWebpackConfig,
       "Css options": actions.replaceWebpackConfig,
+      "Cssoptions commonjs": actions.replaceWebpackConfig,
```
Comment (justinwaite) [inline packages/gatsby-plugin-postcss/src/__tests__/gatsby-node.test.js:114]:
```suggestion
      "Css options commonjs": actions.replaceWebpackConfig,
```

File: packages/gatsby-plugin-postcss/README.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # gatsby-plugin-postcss
 
-Gatsby plugin to handle PostCSS.
+Provides drop-in support for PostCSS
```
Comment (wardpeet) [inline packages/gatsby-plugin-postcss/README.md:3]: I'm unsure what you mean with this? :)

**Review by wardpeet — COMMENTED** [review]
LGTM

File: packages/gatsby-plugin-postcss/README.md (hunk 7 above)
Comment (LekoArts) [reply packages/gatsby-plugin-postcss/README.md:3]: I aligned the heading with the rest of the styling plugins

**Review by LekoArts — APPROVED** [review]
👍

**Comment by lnikell** [comment]
This is will be extremely helpful for migration from v2 to v3, since moving a big project to the new CSS modules exports is complicated and does not have any advantages if the project is well written. Thank you for the fix, you saved me a lot of time :)

**Pull request merged**